### PR TITLE
Attempt to fix flaky test SimpleProducerConsumerTestStreamingDispatcherTest.testRedeliveryFailOverConsumer

### LINKED ISSUE #10117: Flaky-test: SimpleProducerConsumerTestStreamingDispatcherTest.testRedeliveryFailOverConsumer
org.apache.pulsar.broker.service.persistent.SimpleProducerConsumerTestStreamingDispatcherTest is flaky. The testRedeliveryFailOverConsumer test method fails sporadically.

```
java.lang.AssertionError: expected [12] but found [11]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
	at org.testng.Assert.assertEquals(Assert.java:122)
	at org.testng.Assert.assertEquals(Assert.java:907)
	at org.testng.Assert.assertEquals(Assert.java:917)
	at org.apache.pulsar.client.api.SimpleProducerConsumerTest.testRedeliveryFailOverConsumer(SimpleProducerConsumerTest.java:2411)
```

Usage tip: To enable automatic navigation to failure message, open the following links with CTRL/CMD-click.
[example failure 2021-03-31T14:03:07.4363111Z](https://github.com/apache/pulsar/runs/2235404251?check_suite_focus=true#step:8:604)
[example failure 2021-03-31T14:03:07.4275590Z](https://github.com/apache/pulsar/runs/2235404251?check_suite_focus=true#step:8:582)
[example failure 2021-03-30T09:35:08.4204568Z](https://github.com/apache/pulsar/runs/2224991950?check_suite_focus=true#step:8:595)
[example failure 2021-03-29T11:41:50.4073403Z](https://github.com/apache/pulsar/runs/2217711203?check_suite_focus=true#step:8:542)
[example failure 2021-03-28T14:47:42.1363257Z](https://github.com/apache/pulsar/runs/2212657793?check_suite_focus=true#step:8:542)
[example failure 2021-03-28T13:59:15.0401381Z](https://github.com/apache/pulsar/runs/2212348695?check_suite_focus=true#step:8:1414)
[example failure 2021-03-28T13:02:47.3543057Z](https://github.com/apache/pulsar/runs/2212348695?check_suite_focus=true#step:8:603)
[example failure 2021-03-27T16:26:32.6657405Z](https://github.com/apache/pulsar/runs/2208744104?check_suite_focus=true#step:8:1319)

<details>
<summary>Full exception stacktrace</summary>
<code><pre>
java.lang.AssertionError: expected [12] but found [11]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
	at org.testng.Assert.assertEquals(Assert.java:122)
	at org.testng.Assert.assertEquals(Assert.java:907)
	at org.testng.Assert.assertEquals(Assert.java:917)
	at org.apache.pulsar.client.api.SimpleProducerConsumerTest.testRedeliveryFailOverConsumer(SimpleProducerConsumerTest.java:2411)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

</pre></code>
</details>

### PR DESCRIPTION
**Pull request by lhotari**
Fixes #10117 

### Motivation & Modifications

SimpleProducerConsumerTestStreamingDispatcherTest.testRedeliveryFailOverConsumer is one of the most flaky tests.
The main goal of the changes in this PR is to increase the receive timeouts in the testRedeliveryFailOverConsumer . When looking at the test class, the receive timeouts weren't consistent and this PR improves the test class by using constants for receive timeouts. There's now `RECEIVE_TIMEOUT_SECONDS`, `RECEIVE_TIMEOUT_SHORT_MILLIS` and `RECEIVE_TIMEOUT_MEDIUM_MILLIS` constants to be used for 3 different type of use cases of consumer.receive .

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]

**Review by 315157973 — APPROVED** [review]
Good job

**Comment by eolivelli** [comment]
We have flaky tests that are too annoying.
We should disable them asap.
I am going to send a PR tomorrow if no one else does it.
I started a discussion on the dev ML

**Comment by 315157973** [comment]
If you disable it, no one will watch these unit tests anymore, and the coverage of this part will gone. Really annoying

**Comment by eolivelli** [comment]
Please check out my email on dev@
The idea is to disable the 4 offending tests and create blocker issues for the release.
There are already tentative PR for fixes but currently the whole PR merge pipeline is stuck.

This patch is one of those but we cannot merge it is CI does not pass.

If you have a better idea please chime in in discussion on the ML.
I would like to merge this patch and other fixes to flaky tests asap

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java [hunk 1]
```diff
@@ -1124,7 +1127,7 @@ public void testDeactivatingBacklogConsumer(boolean ackReceiptEnabled) throws Ex
         assertTrue(activeSubscriber.contains(sub2));
     }
 
-    @Test(timeOut = 2000)
+    @Test(timeOut = 5000)
```
Comment (merlimat) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java:1130]: I'd say to remove these ad-hoc timeouts and just rely on the default 300 seconds timeout.

**Review by merlimat — APPROVED** [review]

**Pull request merged**